PointInJoint::computeValue()

Hi,

While trying to compile the tests, I get the following error:
```
hpp-constraints/tests/test-jacobians.cc:361:26: error: no matching function for call to ‘hpp::constraints::PointInJoint::computeValue()’
       pij->computeValue ();
```

It seems that in #45,  some `computeValue ();` have been replaced with `computeValue (argument);`; maybe this one was forgotten ?

--- a/src/implicit.cc
+++ b/src/implicit.cc
@@ -76,6 +76,16 @@ namespace hpp {
       }
       if (constantRightHandSide ())
         rhs_ = vector_t ();
+      // TODO currently, I think it is not supported to have an implicit constraint
+      // with function_->outputSize() != function_->outputDerivativeSize()
+      // because we apply the comparison type in the same manner to the value
+      // coeffs and to the Jacobian rows.
+      // EqualToZero should not cause any problem.
+      // Equality is ill-defined coefficient wise (as it is implemented).
+      // Superior and Inferior are (I think) ill-defined.
+      // We should apply the comparison on `function_->value() - rhs`
+      // and on the Jacobian rows, because they have the same dimension.
+      assert (function_->outputSize () == function_->outputDerivativeSize ());
       assert (function_->outputSize () == comparison_.size ());
     }
 
@@ -96,6 +106,8 @@ namespace hpp {
       }
       if (constantRightHandSide ())
         rhs_ = vector_t ();
+      /// TODO See TODO in the constructor above.
+      assert (function_->outputSize () == function_->outputDerivativeSize ());
       assert (function_->outputSize () == comparison_.size ());
     }
 

--- a/src/solver/by-substitution.cc
+++ b/src/solver/by-substitution.cc
@@ -69,7 +69,7 @@ namespace hpp {
       {
         if (contains (nm)) {
           hppDout (error, "Constraint " << nm->functionPtr()->name ()
-                   << " already in " << this->name () << "." << std::endl);
+                   << " already in BySubstitution solver." << std::endl);
           return false;
         }
         ComparisonTypes_t types = nm->comparisonType();
@@ -108,7 +108,7 @@ namespace hpp {
           //      (activeSetFunction(nm->functionPtr(), passiveDofs), types),
           //      segments_t (0), priority);
         }
-        hppDout (info, "Constraints " << name() << " has dimension "
+        hppDout (info, "Constraint has dimension "
                  << dimension());
 
         return true;

--- a/tests/solver-hierarchical-iterative.cc
+++ b/tests/solver-hierarchical-iterative.cc
@@ -159,10 +159,10 @@ BOOST_AUTO_TEST_CASE(one_layer)
 
   solver.add(Implicit::create
              (Orientation::create ("Orientation", device, ee2, tf2),
-              ComparisonTypes_t (6, Equality)), 0);
+              ComparisonTypes_t (3, Equality)), 0);
   solver.add(Implicit::create
              (Position::create    ("Position"   , device, ee2, tf2),
-              ComparisonTypes_t (6, Equality)), 0);
+              ComparisonTypes_t (3, Equality)), 0);
 
   BOOST_CHECK(solver.numberStacks() == 1);
 

--- a/tests/test-jacobians.cc
+++ b/tests/test-jacobians.cc
@@ -358,19 +358,19 @@ BOOST_AUTO_TEST_CASE (SymbolicCalculus_position) {
 
       /// Position
       LiegroupElement value = (*pos) (*q1);
-      pij->computeValue ();
+      pij->computeValue (*q1);
       BOOST_CHECK (pij->value ().isApprox (value.vector()));
       jacobian.setZero ();
       pos->jacobian (jacobian, *q1);
-      pij->computeJacobian ();
+      pij->computeJacobian (*q1);
       BOOST_CHECK (pij->jacobian ().isApprox (jacobian));
       // Relative position
       value = (*relpos) (*q1);
-      relpos_sb_ptr->computeValue ();
+      relpos_sb_ptr->computeValue (*q1);
       BOOST_CHECK (relpos_sb_ptr->value ().isApprox (value.vector()));
       jacobian.setZero ();
       relpos->jacobian (jacobian, *q1);
-      relpos_sb_ptr->computeJacobian ();
+      relpos_sb_ptr->computeJacobian (*q1);
       BOOST_CHECK (relpos_sb_ptr->jacobian ().isApprox (jacobian));
   }
 }

